plugins/plugin-chart-echarts/src/Gauge/controlPanel.tsx

```diff
@@ -0,0 +1,287 @@
+              renderTrigger: true,
+              min: 10,
+              max: 25,
+              default: DEFAULT_FORM_DATA.fontSize,
+            },
+          },
+        ],
+        [
+          {
+            name: 'animation',
+            config: {
+              type: 'CheckboxControl',
+              label: t('Animation'),
+              description: t(
+                'Whether to animate the progress bar and the value or just display them',
+              ),
+              renderTrigger: true,
+              default: DEFAULT_FORM_DATA.animation,
+            },
+          },
+        ],
+        [<h1 className="section-header">{t('Progress Bar')}</h1>],
+        [
+          {
+            name: 'roundcap',
+            config: {
+              type: 'CheckboxControl',
+              label: t('Roundcap'),
+              description: t('Style the ends of the progress bar with a roundcap'),
+              renderTrigger: true,
+              default: DEFAULT_FORM_DATA.roundcap,
+            },
+          },
+        ],
+        [
+          {
+            name: 'show_axis_tick',
+            config: {
+              type: 'CheckboxControl',
+              label: t('Show axis line ticks'),
+              description: t('Whether to show minor ticks on the axis'),
+              renderTrigger: true,
+              default: DEFAULT_FORM_DATA.showAxisTick,
+            },
+          },
+        ],
+        [
+          {
+            name: 'show_split_line',
+            config: {
+              type: 'CheckboxControl',
+              label: t('Show split lines'),
+              description: t('Whether to show the split lines on the axis'),
+              renderTrigger: true,
+              default: DEFAULT_FORM_DATA.showSplitLine,
+            },
+          },
+        ],
+        [
+          {
+            name: 'split_number',
+            config: {
+              type: 'SliderControl',
+              label: t('Split number'),
+              description: t('Number of split segments on the axis'),
+              renderTrigger: true,
+              min: 3,
+              max: 30,
+              default: DEFAULT_FORM_DATA.splitNumber,
+            },
+          },
+        ],
+        [
+          {
+            name: 'start_angle',
+            config: {
+              type: 'TextControl',
+              label: t('Start angle'),
+              description: t('Angle at which to start progress axis'),
+              renderTrigger: true,
+              default: DEFAULT_FORM_DATA.startAngle,
+            },
+          },
+        ],
+        [
+          {
+            name: 'end_angle',
+            config: {
+              type: 'TextControl',
+              label: t('End angle'),
+              description: t('Angle at which to end progress axis'),
+              renderTrigger: true,
+              default: DEFAULT_FORM_DATA.endAngle,
+            },
+          },
+        ],
```

Comment:
Imho, [most gauges](http://www.agcoauto.com/content/images/electrical/typical_dash_gauges.jpg) are likely to be vertically symmetrical, with the “empty” at the bottom (think speedometer). Thus, we could save one parameter by only specifying something like “Gauge angle/aperture”. The input could be a [0, 360] slider (I'd say stepped every 5 or 10°).
We can then compute:
* `start_angle` = 90 + `gauge_angle` ÷ 2
* `end_angle` = 90 − `gauge_angle` ÷ 2
(apparently, there is no need for modulo)


… yet at the same time, it makes more sense to make 360°/[ring gauges](https://echarts.apache.org/examples/en/editor.html?c=gauge-ring) with the start on top.

Reply:
P.S.

I'm here thinking how to improve UX/reduce user's cognitive load. (A proper UX designer would be more adequate to advise on this, though.)

I eventually believe one could be even more prescriptive, and hardcode start and end angle. (E.g. at 225 and −45 deg, respectively, as in [the basic example](https://echarts.apache.org/examples/en/editor.html?c=gauge-simple)).
Or these could be preset, but modifiable in a section like Explore/Customize/Advanced charts options — possibly still through one unique slider. (Off-topic: accessing this section could be linked to ad hoc privileges — i.e. to prevent “basic” users to tinker with it and decrease charts homogeneity.)

This chart type could be for < 360° gauges only, vertically symmetrical, with one unique data point visualized. Then, one could define _a separate_ chart type for [“ring gauges”](https://echarts.apache.org/examples/en/editor.html?c=gauge-ring). This type would have `startAngle` and `endAngle` equal to 90° and −270°, respectively (i.e. clockwise, starting and ending at noon) _and could display_ one or multiple _data point(s)_. Multiplying chart types isn't ideal, but I guess that in this case, it could simplify UI and actually address to not-entirely-overlapping use cases.